debugging-errors.mdx

```diff
@@ -0,0 +1,120 @@
+to optimize resource configuration for your situation!
+
+### Query Timeout / Unclear Failure
+
+If you are running a query from a notebook or somewhere locally, the polling for query status might have a shorter
+timeout, so if the query times out before completion, we'd recommend verifying whether the query is still running in the
+console by looking at the query run page and validating that there is a pod in the cluster running the query. Otherwise
+your query may have also completed after the polling timeout, in which case you can also view query status in the
+console.
+
+### Please double check that all features have resolvers and resolvers do not have circular dependencies.
+
+The first thing to validate is that all features have resolvers. If all features have resolvers, then you might see
+this error message if Chalk is unable to construct a dependency graph for resolving the features that you are
+querying. This is usually a result of a join that cannot be processed. Please identify which feature(s) is triggering
+the circular dependency error and reach out to us with this context!
+
+### Duplicate Resolver Shortname
+
+The first thing to check is whether there actually are duplicate resolvers by the same name. If there are not, this
+error message can also sometimes be an indicator of a missing import.
+
+## `chalk apply` Errors
+
+### branch server might still be starting
+
+You would usually encounter this error when running `chalk apply --branch`. In the Chalk console under `Branches`, you
+can check the status of the branch server. If you are concerned that there are lingering processes on branches, such
+as queries on branches that you would like to kill, you can also stop and restart the branch server in the console to
+kill any processes in progress on the branch server.
+
+### Unable to load dashboards after deploy
+
+This usually means that there was an error on the Chalk servers in parsing the code uploaded in the latest deploy. A
+common cause of this is if you deploy code with `ChalkClient.query()` or `ChalkClient.offline_query()` in your code
+that is not nested within another function.
+
+For example, say you have a file like this within your source code that you deploy to Chalk
+
+```python
+from chalk import ChalkClient
+
+ChalkClient().offline_query(
+    input={User.id: [1]},
+    output=[User.sum_transactions],
+    store_plan_stages=True,
+    recompute_features=True
+).resolver_replay(sum_transaction_amount)
+```
+
+You would instead want to wrap the query in a function so that Chalk does not try to parse the query code as a
+function or resolver.
+
+```python
+from chalk import ChalkClient
+
+if __name__=='__main__':
+    ChalkClient().offline_query(
+        input={User.id: [1]},
+        output=[User.sum_transactions],
+        store_plan_stages=True,
+        recompute_features=True
+    ).resolver_replay(sum_transaction_amount)
+```
```

Comment:
I have PR out to make this impossible btw, so maybe hold off on this particular reason

Reply:
omg ok holding